### ISSUE TITLE
Test debugging documentation insufficient

### ISSUE THREAD

**Issue by garthk** [Microsoft/vscode-go#183]
> Once this is installed, go to the Code debug viewlet and select the configuration gear, placing the following in your launch.json:

> ```
> {
>     "version": "0.1.0",
>     "configurations": [
>         {
>             "name": "Launch main.go",
>             "type": "go",
>             "request": "launch",
>             "mode": "debug",
>             "program": ".",
>             "env": {},
>             "args": []
>         }
>     ]
> }
> ```

The default `launch.json` has `"version": "0.2.0"` and `"program": "${workspaceRoot}/."` but so far, so good.

> The program option can refer to a package folder to debug, or a file within that folder.
>
> The mode parameter can be set to:
>
> * `debug` to compile the contents of the program folder and launch under the debugger. [default]
> * `test` to debug tests in the program folder.
> * `exec` to run a pre-built binary instead of building the current code in the program folder.

Is that it? Let's find out. Set `"mode": "test"`, hit F5…

```
2016/01/28 10:51:24 debugger.go:55: launching process with args: [./z.test]
could not launch process: stat /Users/x/go/src/github.com/y/z/z.test: no such file or directory
```

… and hit a wall, with many questions for which the documentation lacks answers.

* What process couldn't be launched? With what arguments? `dlv` is in `$GOPATH/bin`, and that's on the `$PATH`; what's this `z.test` file with its name copied from my project directory supposed to be?
* What else do I need to set up?
* What troubleshooting steps could I run?
* I see `console.log` calls in `src/debugAdapter/goDebug.ts`; how do I get to that output?
* If my testable stuff is in `lib`, should I set `"program": "${workspaceRoot}/lib"`? That changes the `args` in the debug console to `[./lib.test]`, but the next line now reads `could not launch process: could not fork/exec`; why?

**Comment by garthk** [comment]
I also tried `"mode": "debug"` and `"program": "${workspaceRoot}/thebinary", and added an entry to `args`. Now the error is:

```
2016/01/28 11:06:58 debugger.go:55: launching process with args: [./debug theargument]
could not launch process: could not fork/exec
```

… and I'm accumulating quite a collection of zombied `dlv` processes which remain even after I quit Code.

**Comment by lukehoban** [comment]
> The default launch.json has "version": "0.2.0" and "program": "${workspaceRoot}/." but so far, so good.

Fixed in ce0c60709bfdbad8501b8c65ea64d536aca4225d.

> `could not launch process: could not fork/exec`

This error message normally means that you haven't signed your `dlv` binary.  See https://github.com/derekparker/delve/wiki/Building.

> `could not launch process: stat /Users/x/go/src/github.com/y/z/z.test: no such file or directory`

I haven't seen this before, but I'll bet it's also due to not having a signed binary.  I confirmed that `"mode": "test"` still works as expected in a few projects on my machine.

As for the unfortunate need to sign the `dlv` binary manually - I hope we can eventually take care of that - but it will require building and signing a dlv binary as part of the `vscode-go` project and deploying that dependency with the extension.  Several pieces of infrastructure needed before we'll be able to do that.  So for now we just document the installation requirements.
